beacon_node/beacon_chain/src/block_verification_types.rs

```diff
@@ -65,8 +65,22 @@ impl<E: EthSpec> RpcBlock<E> {
         block: Arc<SignedBeaconBlock<E>>,
         blobs: Option<BlobSidecarList<E>>,
     ) -> Result<Self, AvailabilityCheckError> {
-        if let Some(blobs) = blobs.as_ref() {
-            data_availability_checker::consistency_checks(&block, blobs)?;
+        if let (Some(blobs), Ok(block_commitments)) = (
+            blobs.as_ref(),
+            block.message().body().blob_kzg_commitments(),
+        ) {
+            if blobs.len() != block_commitments.len() {
+                return Err(AvailabilityCheckError::MissingBlobs);
+            }
+            for (blob, &block_commitment) in blobs.iter().zip(block_commitments.iter()) {
+                let blob_commitment = blob.kzg_commitment;
+                if blob_commitment != block_commitment {
```

Comment:
Nice call on dropping the other checks, much cleaner 👍 
Although, should we be checking the `block_root` key matches? The potential issue I'm worried about is a peer sending us block for slot X and blobs from slot Y. We will end up storing the blob for slot X as the blobs for slot Y which might give inconsistent results on http queries and responding for other peer requests over rpc.

Reply:
Added here, piped block root through a lot to try to make sure we gather metrics for block root calcs and avoid calculating it twice https://github.com/sigp/lighthouse/pull/4732/commits/220fda1f2968df50ad90b5b6e1390c974853b2f0